### ISSUE TITLE
Machine learning model

### ISSUE THREAD

**Issue by arjun-krishna1** [arjun-krishna1/santatron#7]
# Clustering
- [x] find api to get similarity of two strings
- [x] set up and use to find similarity between "tomato juice" and "potato pancake"
- [x] package into function that will take two strings and return their similarity
- [x] create a function that will take lists of words and find the distance of each word to every other word
- [x] decide on some metric for choosing amazon search keywords
  - [x] total distance to all other words, minimum distance to a group of other keywords most similar to it
- [x] create a function that will take lists of words and return 3 words based on this model that is most likely to find a gift that is coherent

**Comment by Joe-Joe-Joe-Joe** [comment]
- pool of keywords (variable length)
- each person contributing to the gift pool inputs 5 keywords
- encoder takes up to 20 keywords from pool of keywords and encodes then down to 5
- those 5 kewords form the search query to google/amazon/etc

**Comment by Joe-Joe-Joe-Joe** [comment]
dec 24
- find keywords training dataset (2:30)
- write code for encoder and decoder models (7:30)
- serialize models, training script, data interface from database (8:30)

dec 25
- finish training models
- implement amazon search API
- implement smart contract system OR paypal API invoices

**Comment by arjun-krishna1** [comment]
3 words that are the most similar to the suggested keywords by gift pool donors
doesn't have to be in the suggested keywords (any english word)
have to be unique
append "gift", that's the amazon search query

**Comment by arjun-krishna1** [comment]
joe finished this
